plugins/inputs/github/github.go

```diff
@@ -0,0 +1,168 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"sync"
+	"time"
+
+	gh "github.com/google/go-github/github"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"golang.org/x/oauth2"
+)
+
+// GitHub - plugin main structure
+type GitHub struct {
+	Repositories []string          `toml:"repositories"`
+	AccessToken  string            `toml:"access_token"`
+	HTTPTimeout  internal.Duration `toml:"http_timeout"`
+	githubClient *gh.Client
+}
+
+const sampleConfig = `
+  ## Specify a list of repositories
+  repositories = ["influxdata/influxdb"]
+
+  ## API Key for GitHub API requests
+  api_key = ""
+
+  ## Timeout for GitHub API requests
+  http_timeout = "5s"
+`
+
+// NewGitHub returns a new instance of the GitHub input plugin
+func NewGitHub() *GitHub {
```

Comment:
Doesn't look like this is used.  I would probably not use it and set the HTTPTimeout in the `init()` function.